assets/js/blocks/cart-checkout/cart/full-cart/index.tsx

```diff
@@ -114,6 +114,14 @@ const Cart = ( { attributes }: CartProps ) => {
 		cart,
 	};
 
+	const discountsSlotFillProps = {
+		extensions,
+		cart,
+		components: {
+			ValidatedTextInput,
+		},
```

Comment:
I'm still not sold on this `components` pattern tbh. Why are we doing it this way? Do people need this exact input with its validation context and all? can't just they get a regular input and handle validation themselves?

Reply:
I guess I got too bogged down here, I think there's actually no reason to pass this component through... let me get rid of it and see how it goes 👌🏼 .

Reply:
Just noting that Thomas and I chatted about this today. One of the reasons why this was passed through was to:

- Expose (what would likely be a common type of) component for extension developers to use to preserve consistent styling and behaviour.
- Make it easier for extension developers to implement this kind of functionality.

However, the tradeoffs are that we are establishing a contract to preserve this component and kicks off a pattern that might be hard to maintain in the future if extension developers have need for other types of components in the checkout (checkbox etc). There's some element here of just including this component because Points & Rewards needs it.

So for now, we agreed that we'll hold off exposing this and Thomas will implement the functionality directly in Points & Rewards. I do think we'll still need to address the issue of exposing some components/style system for extension developers to use so they can inherit/preserve the current (and any evolution of the) design of the blocks which makes for a better shopper experience. The work Thomas will be doing in Points and Rewards implementation will unblock that integration but will be more inherently fragile in relation to potential changes in the Cart and Checkout design down the road (for example when we potentially start including Global Styles from Gutenberg to allow for more customization options by merchants etc). So I think we will need to address this in a followup.

Reply:
> Expose (what would likely be a common type of) component for extension developers to use to preserve consistent styling and behaviour.
> Make it easier for extension developers to implement this kind of functionality.

Why wouldn't having the component in `@woocommerce/blocks-checkout` be enough? previously we passed shippingSelector from there to preserve an already passed context I guess, but that pattern is up for reconsideration during this cooldown in which we might delete the shipping methods slotFill for a registration API.

In my mind, `components` was used to pass components that are harder to move to the checkout package or are very opinionated to a specific case.

I'm not against exposing components in general via a components package, in that case, we have a real boundary of "all code in this folder is public" type of protection, making it easier for us to maintain it.

Reply:
This is exactly the sort of discussion that we need to work through separately now that we are starting to see integrations that could use more of these components (whether published via package, or passed through as an interface prop etc).

Reply:
I already tried moving the `ValidatedTextInput` component to the checkout package, but it didn't work with the validation context, because they were in different packages it seemed they couldn't work together. Then we tried passing the context through the slot, and that worked, but wasn't the right approach to take as we didn't want to expose this context to third parties.

For now I'm sticking with what Darren has already said - we will duplicate the component in P&R until we can revisit this and find a way to expose `ValidatedTextInput` to extensions.